specs/phase0/validator.md

```diff
@@ -391,7 +393,13 @@ def get_block_signature(state: BeaconState, block: BeaconBlock, privkey: int) ->
 
 A validator is expected to create, sign, and broadcast an attestation during each epoch. The `committee`, assigned `index`, and assigned `slot` for which the validator performs this role during an epoch are defined by `get_committee_assignment(state, epoch, validator_index)`.
 
-A validator should create and broadcast the `attestation` to the associated attestation subnet when either (a) the validator has received a valid block from the expected block proposer for the assigned `slot` or (b) one-third of the `slot` has transpired (`SECONDS_PER_SLOT / 3` seconds after the start of `slot`) -- whichever comes _first_.
+For each `slot`, a validator must generate a uniform random variable `slot_timing_entropy` between `(-SECONDS_PER_SLOT / ATTESTATION_ENTROPY_DIVISOR, SECONDS_PER_SLOT / ATTESTATION_ENTROPY_DIVISOR)` with millisecond resolution and using local entropy.
```

Comment:
can the entropy be shared by multiple validators that is served under the same beacon node? 

this will simplify the beacon node implementation

Reply:
Yes, the entropy can be shared by multiple validators that are served under the same beacon node.

Reply:
Do you mean the source of entropy can be shared or that randomly selected `slot_timing_entropy` value can be shared by all validators served by the same beacon node?  

There are significant performance implications of every individual validator has to select the latest head and create its attestation at a different time.  Currently a Validator Client only needs to ask the beacon node to create at most one `AttestationData` per committee per slot because all validators in that same committee can create an attestation from that `AttestationData`. And all validators can share the same selected head block.

With this change, if the value of `slot_timing_entropy` can't be shared, the number of validators a beacon node could support would be significantly reduced as it would need to update fork choice and create a new `AttestationData` for each individual validator.

Reply:
This is the gist of this fix:
> By making the attestation production time unpredictable to the attacker & unique for each validator, ...

The attestation production time doesn't _have_ to be unique for each and every validator. However, it is absolutely crucial that the attestation production time is unpredictable for anyone who does not control the validator, and/or the beacon node (for clients where the beacon node is the driver of validator duties). So, validators served by the same beacon node can have the same attestation production time, i.e., they can share the source of the entropy **and** the actual `slot_timing_entropy` value.